ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -207,31 +191,32 @@ azure_flush(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session, WT_FILE_SYST
     WT_UNUSED(storage_source);
     WT_UNUSED(config);
     azure_file_system *azure_fs = reinterpret_cast<azure_file_system *>(file_system);
-    WT_FILE_SYSTEM *wt_filesystem = azure_fs->wt_fs;
+    WT_FILE_SYSTEM *wtFileSystem = azure_fs->wt_fs;
 
-    std::string src_path = azure_path(azure_fs->home_dir, source);
-    bool exists_native = false;
+    if (!std::filesystem::exists(object)) {
+        std::cerr << "azure_flush: Object: " << object << " does not exist." << std::endl;
+        return ENOENT;
+    }
 
-    int ret = wt_filesystem->fs_exist(wt_filesystem, session, src_path.c_str(), &exists_native);
+    bool exists_native = false;
+    int ret = wtFileSystem->fs_exist(
+      wtFileSystem, session, std::filesystem::canonical(source).string().c_str(), &exists_native);
     if (ret != 0) {
-        std::cerr << "azure_flush: Failed to check for the existence of " + std::string(source) +
-            " on the native filesystem."
-                  << std::endl;
+        std::cerr << "azure_flush: Failed to check for the existence of " << source
+                  << " on the native filesystem.";
```

Comment:
The `std::endl` has disappeared - looks like a mistake. There's another spot below with the same issue.

Reply:
oops - yes this was a mistake. Amended now: https://github.com/wiredtiger/wiredtiger/pull/8763/commits/607112cb4257befc1b438c807c53b2445cb8f08c